zkvm/src/signer/mod.rs

```diff
@@ -0,0 +1,68 @@
+#![allow(non_snake_case)]
+
+use crate::transcript::TranscriptProtocol;
+use curve25519_dalek::ristretto::RistrettoPoint;
+use curve25519_dalek::scalar::Scalar;
+use merlin::Transcript;
+use rand;
+
+// Modules for signing protocol
+
+pub mod prover;
+
+pub struct PrivKey(Scalar);
+pub struct PubKey(RistrettoPoint);
+pub struct MultikeyWitness(Vec<PubKey>); // TODO: also include Option<Scalar> for signing key?
+pub struct PubKeyHash(Scalar);
```

Comment:
do we need this type throughout the protocol?

Reply:
I got rid of the `PubKeyHash` type, now that we just have this as a function call over `Multikey`.